src/main/java/core/basesyntax/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(Storage storage, FruitTransaction transaction) {
+        int currentQuantity = storage.getQuantity(transaction.getFruit());
+        int newQuantity = currentQuantity - transaction.getQuantity();
+
+        if (newQuantity < 0) {
+            throw new RuntimeException("Insufficient stock for purchase of "
+                    + transaction.getFruit());
+        }
+
+        storage.updateFruit(transaction.getFruit(), newQuantity);
+    }

```

Comment:
The `PurchaseOperation` class seems to correctly implement the logic for handling purchase operations by updating the storage after checking for sufficient stock. However, it is crucial to ensure that the `apply` method adheres to the encapsulation principle. Consider if all elements within the class that should be private are indeed private to prevent unwanted access.